tools/create_service_diagnostics_bundle.py

```diff
@@ -0,0 +1,225 @@
+        type=str,
+        required=True,
+        default=None,
+        help="The service name to create the bundle for",
+    )
+
+    parser.add_argument(
+        "--bundles-directory",
+        type=str,
+        required=True,
+        default=None,
+        help="The directory where bundles will be written to",
+    )
+
+    parser.add_argument(
+        "--yes",
+        action="store_true",
+        help="Disable interactive mode and assume 'yes' is the answer to all prompts.",
+    )
+
+    return parser.parse_args()
+
+
+def preflight_check() -> (int, bool, dict):
+    args = parse_args()
+    package_name_given = args.package_name
+    service_name = args.service_name
+    bundles_directory = args.bundles_directory
+    should_prompt_user = not args.yes
+
+    (is_authenticated, message) = is_authenticated_to_dcos_cluster()
+    if not is_authenticated:
+        log.error(
+            "We were unable to verify that you're authenticated to a DC/OS cluster.\nError: %s",
+            message,
+        )
+        return (1, False, {})
+
+    (rc, cluster_or_error) = attached_dcos_cluster()
+    if rc != 0:
+        log.error(
+            "We were unable to verify the cluster you're attached to.\nError: %s",
+            service_name,
+            cluster_or_error,
+        )
+        return (rc, False, {})
+
+    (cluster_name, dcos_version, cluster_url) = cluster_or_error
+
+    (rc, marathon_app_or_error) = get_marathon_app(service_name)
+    if rc == 0:
+        package_name = marathon_app_or_error.get("labels", {}).get("DCOS_PACKAGE_NAME")
+        package_version = marathon_app_or_error.get("labels", {}).get("DCOS_PACKAGE_VERSION")
+    else:
+        log.info(
+            "We were unable to get details about '%s'.\nIssue: %s",
+            service_name,
+            marathon_app_or_error,
+        )
+        log.info(
+            "Maybe the '%s' scheduler is not running. That's ok, we can still try to fetch any "
+            + "artifacts related to it",
+            service_name,
+        )
+        package_name = package_name_given
+        package_version = "n/a"
+
+    if package_name_given != package_name:
+        log.error(
+            "Package name given '%s' is different than actual '%s' package name: '%s'",
+            package_name_given,
+            service_name,
+            package_name,
+        )
+        log.info("Try '--package-name=%s'", package_name)
+        return (1, False, {})
+
+    return (
+        0,
+        True,
+        {
+            "package_name": package_name,
+            "service_name": service_name,
+            "package_version": package_version,
+            "cluster_name": cluster_name,
+            "bundles_directory": bundles_directory,
+            "dcos_version": dcos_version,
+            "cluster_url": cluster_url,
+            "should_prompt_user": should_prompt_user,
+        },
+    )
+
+
+def main(argv):
+    (rc, should_proceed, args) = preflight_check()
```

Comment:
You don't need the parentheses.

Reply:
ad4836c093791a9d53f728b08c23a3b0e2f03778